src/ImageDownloader.ts

```diff
@@ -0,0 +1,35 @@
+import sha1 from 'js-sha1';
+import { getSwarmGatewayUrl } from './Swarm';
+import { FileSystem } from './FileSystem';
+
+const getExtensionFromPath = (path: string): string => {
```

Comment:
I think we have quite a few places now which deal with path and filenames, maybe it's time to be consolidated in one place

Reply:
I agree and I think the FileSystem could be a good place for that.